Add hints to GUI, clean up some internal code

dagerikhl opened the pull request:
Got a bit more advanced than I expected, but I think the result turned out quite nicely :D 
Having to use a fetch to get the hints is not optimal, but I was unable to use a load function inside the Markdown components, so adding the hints to the session was not an option, not one I was able to solve at least.

palexcast commented:
I love it! Looks really nice and works well! Also cowsay <3 
 But there seems to be an issue with other code blocks now:
 
<img width="507" alt="image" src="https://user-images.githubusercontent.com/15246162/158695852-6873dd11-b3cb-41ca-84dc-4e5d3d1040e5.png">

They seem to be newlined

palexcast commented:
Just a proposal from how i planned to solve stuff in my branch.

Instead of rendering the hint on each codespan element, I had a separate "hint" element in the layout.svelte which was hidden.
I then used a hint store to trigger the hint with an id/key from each codespan element.

dagerikhl commented:
I should have seen that, sorry, tunnel vision :O 

I've fixed the wrapping issue, it was as you said, and issue with the container.

And your suggestion to usa a store was brilliant! I've rewritten it to such a structure, setting the hint key in a store, and subscribing to it in an overlay component in layout. Before, I did a fetch on `/hints.js` on each codespan, which was suboptimal at best. Now I only have to fetch it once, brilliant! Thanks for the suggestion :D 

Let me know what you think of the new solution :)

palexcast commented on src/lib/_workshop-internals/components/hint-overlay/HintOverlay.svelte:
```diff
@@ -0,0 +1,76 @@
+
+{#if hint}
+	<div class="hint-container" on:click={handleCloseHint}>
+		<div class="rpgui-container framed hint" on:click|stopPropagation>
```
I think the stopProgagation i stopping the user from selecting text. Might be an issue is there is something you would want to copy. Not very important, just worth noting

dagerikhl replied:
Actually, it was the rpgui framework adding `user-select: none` to all containers, but nice catch! Fixed it so it's possible to select text.
(The `stopPropagation` is to prevent clicking the box from closing the hint, but clicking outside it will close the hint.)

palexcast commented on src/lib/_workshop-internals/components/hint-overlay/HintOverlay.svelte:
```diff
@@ -0,0 +1,76 @@
+	let hints: Hints;
+
+	hintStore.subscribe((hintKey) => {
+		console.log({ hintKey, hints });
```
Rm console.log

dagerikhl replied:
Thank you :D

palexcast reviewed (approved):
Nice job! Looks really good! 
Added two comments, but nothing important.
